src/PostgREST/DbStructure.hs

```diff
@@ -311,22 +313,68 @@ accessibleTables =
       relname as table_name,
       d.description as table_description,
       (
-        c.relkind in ('r', 'v', 'f')
-        and (pg_relation_is_updatable(c.oid::regclass, false) & 8) = 8
-        -- The function `pg_relation_is_updateable` returns a bitmask where 8
-        -- corresponds to `1 << CMD_INSERT` in the PostgreSQL source code, i.e.
-        -- it's possible to insert into the relation.
-        or (exists (
-          select 1
-          from pg_trigger
-          where
-            pg_trigger.tgrelid = c.oid
-            and (pg_trigger.tgtype::integer & 69) = 69)
-            -- The trigger type `tgtype` is a bitmask where 69 corresponds to
-            -- TRIGGER_TYPE_ROW + TRIGGER_TYPE_INSTEAD + TRIGGER_TYPE_INSERT
-            -- in the PostgreSQL source code.
+        (
+          has_table_privilege(c.oid, 'INSERT')
+          or has_any_column_privilege(c.oid, 'INSERT')
+        )
```

Comment:
I'm not sure about adding those lines here, either. I think this would only affect OpenAPI output anyway - but I'm not even sure in which way. I had a quick glance at `OpenApi.hs` but didn't understand immediately what was going on with the insertable flag.

I suggest to leave any changes to OpenApi out of this PR and focus on the `OPTION` request's `Allow` header (see other comment).

If you decide to make this query similar to the `allTables` query, there's also a nice possibility to refactor:
What about making `accessibleTables` just take the query from `allTables` and then adding the two `WHERE` conditions, e.g. via argument or so? Would remove a lot of duplicated code here. Kind of similar to what `accessibleProcs` does.

Reply:
> I suggest to leave any changes to OpenApi out of this PR and focus on the OPTION request's Allow header (see other comment).

OK, I get the idea now. The issue mentioned the problem communicating the Api spec using Swagger though, that's why I tried modifying OpenApi too.

> What about making accessibleTables just take the query from allTables and then adding the two WHERE conditions

Seems good, I'll do that.

Reply:
> OK, I get the idea now. The issue mentioned the problem communicating the Api spec using Swagger though, that's why I tried modifying OpenApi too.

Background regarding OpenAPI: Every change we make here now is basically a waste of time, because want to split this off into `postgrest-contrib` (to be created) as SQL, see #1698.

Reply:
> If you decide to make this query similar to the allTables query, there's also a nice possibility to refactor:
What about making accessibleTables just take the query from allTables and then adding the two WHERE conditions, e.g. via argument or so? Would remove a lot of duplicated code here. Kind of similar to what accessibleProcs does.

The refactor result won't be as smooth as `accessibleProcs`, `accessibleTables` has an extra Join to `pg_catalog.pg_description` and uses the description field [here](https://github.com/PostgREST/postgrest/blob/main/src/PostgREST/DbStructure.hs#L330), in `allTables` it's a [NULL value](https://github.com/PostgREST/postgrest/blob/main/src/PostgREST/DbStructure.hs#L471). 

Would it be OK for that extra JOIN to be used by `allTables` too? Or should it be conditionally added to `accessibleTables` only?

Reply:
I was about to say that it's no problem to add the description to the `allTables` query, because it will go away completely once we remove OpenApi from core. But the same holds throw for the whole `accessibleTables` query. So... it probably doesn't make too much sense to actually refactor it, when we will remove it soon.

Reply:
Makes sense. Will leave it as it is.